### PR TITLE
Include and set ProGuard uuid in Flutter Options

### PR DESCRIPTION
**Pull request by Wcisang**
## :scroll: Description

I just added a proGuardUuid property to the Options and passed it to Sentry's android native initialization.

## :bulb: Motivation and Context

Due to the flutter SDK not having support to configure the proguard uuid, it is necessary to perform a large workaround to perform this action.
So this simple modification already makes it possible for Flutter apps to configure the proguard uuid at runtime.

## :green_heart: How did you test it?

created unit tests for testing defaults and custom values

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: dart/lib/src/sentry_options.dart [hunk 1]
```diff
@@ -279,6 +279,9 @@ class SentryOptions {
   /// If enabled, [scopeObservers] will be called when mutating scope.
   bool enableScopeSync = true;
 
+  /// Sets the ProGuard uuid for Android platform.
+  String? proGuardUuid;
```
Comment (marandaneto) [inline dart/lib/src/sentry_options.dart]: Add to `SentryFlutterOptions` instead.

Comment (marandaneto) [inline dart/lib/src/sentry_options.dart]:
```suggestion
  /// Sets the ProGuard uuid for Android platform.
  String? proguardUuid;
```

Let's align on the naming.

**Comment by marandaneto** [comment]
@Wcisang you could technically set it directly into the [Manifest file on Android](https://docs.sentry.io/product/cli/dif/#proguard-mapping-upload).
Does that help?

**Comment by Wcisang** [comment]
> 

@marandaneto 
Because we use a protection tool that runs after the build in the pipeline and the obfuscation map changes completely every run. And with that, we need to set this configuration at runtime.

File: dart/lib/src/sentry_options.dart (hunk 1 above)
Comment (Wcisang) [reply dart/lib/src/sentry_options.dart]: done! thanks!

Comment (Wcisang) [reply dart/lib/src/sentry_options.dart]: done! thanks!

**Review by marandaneto — APPROVED** [review]
Thanks @Wcisang 
Before merging, please add an entry to our changelog :)

**Comment by Wcisang** [comment]
> Thanks @Wcisang Before merging, please add an entry to our changelog :)

Done!
When does the next release happen?

**Comment by marandaneto** [comment]
> > Thanks @Wcisang Before merging, please add an entry to our changelog :)
> 
> Done! When does the next release happen?

I can cut a release once CI is happy, just fixed a detekt issue.

**Pull request merged**